util/codegen/codegen.go

```diff
@@ -263,3 +267,49 @@ func IsViewType(typ types.Type) bool {
 	}
 	return t.Field(0).Name() == "ж"
 }
+
+// MaxCopyrightYear reports the greated copyright year in non-generated *.go
+// files in the current directory.
+//
+// It panics on I/O error, as it's assumed this is only being used by "go
+// generate" or GitHub actions.
+func MaxCopyrightYear(dir string) (year int) {
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		panic(err)
+	}
+	rxYear := regexp.MustCompile(`^// Copyright \(c\) (20\d{2}) `)
+	rxGenerated := regexp.MustCompile(`^// Code generated .* DO NOT EDIT\.$`)
+Files:
+	for _, f := range files {
+		name := f.Name()
+		if f.IsDir() ||
+			!strings.HasSuffix(name, ".go") ||
+			strings.HasSuffix(name, "_clone.go") ||
+			strings.HasSuffix(name, "_view.go") ||
+			strings.HasSuffix(name, "_test.go") {
+			continue
+		}
+		src, err := os.ReadFile(filepath.Join(dir, name))
+		if err != nil {
+			panic(err)
+		}
+		bs := bufio.NewScanner(bytes.NewReader(src))
+		for bs.Scan() {
+			line := bs.Bytes()
+			if m := rxYear.FindSubmatch(line); m != nil {
+				if y := must.Get(strconv.Atoi(string(m[1]))); y > year {
```

Comment:
I think the usual guidance is the year of creation, and then not updating date in the future even as changes are made in the code. That would suggest the earliest non-zero year, instead of the latest.

However anything would be better than the tree being broken, we can discuss it more this coming week.

Reply:
Renamed the func to "CopyrightYear" instead of "MaxCopyrightYear" and added a TODO and changed the commit to not "Fixes" the issue but rather only "Updates".